src/main/java/picard/cmdline/CommandLineProgram.java

```diff
@@ -65,45 +80,57 @@
+    private static Boolean useLegacyParser;
+
+    @Argument(common=true, optional=true)
     public List<File> TMP_DIR = new ArrayList<File>();
 
-    @Option(doc = "Control verbosity of logging.", common=true)
+    @Argument(doc = "Control verbosity of logging.", common=true)
     public Log.LogLevel VERBOSITY = Log.LogLevel.INFO;
 
-    @Option(doc = "Whether to suppress job-summary info on System.err.", common=true)
+    @Argument(doc = "Whether to suppress job-summary info on System.err.", common=true)
     public Boolean QUIET = false;
 
-    @Option(doc = "Validation stringency for all SAM files read by this program.  Setting stringency to SILENT " +
+    @Argument(doc = "Validation stringency for all SAM files read by this program.  Setting stringency to SILENT " +
             "can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) " +
             "do not otherwise need to be decoded.", common=true)
     public ValidationStringency VALIDATION_STRINGENCY = ValidationStringency.DEFAULT_STRINGENCY;
 
-    @Option(doc = "Compression level for all compressed files created (e.g. BAM and GELI).", common=true)
+    @Argument(doc = "Compression level for all compressed files created (e.g. BAM and GELI).", common=true)
     public int COMPRESSION_LEVEL = BlockCompressedStreamConstants.DEFAULT_COMPRESSION_LEVEL;
 
-    @Option(doc = "When writing SAM files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort a SAM file, and increases the amount of RAM needed.", optional=true, common=true)
+    @Argument(doc = "When writing SAM files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort a SAM file, and increases the amount of RAM needed.", optional=true, common=true)
     public Integer MAX_RECORDS_IN_RAM = SAMFileWriterImpl.getDefaultMaxRecordsInRam();
 
-    @Option(doc = "Whether to create a BAM index when writing a coordinate-sorted BAM file.", common=true)
+    @Argument(doc = "Whether to create a BAM index when writing a coordinate-sorted BAM file.", common=true)
     public Boolean CREATE_INDEX = Defaults.CREATE_INDEX;
 
-    @Option(doc="Whether to create an MD5 digest for any BAM or FASTQ files created.  ", common=true)
+    @Argument(doc="Whether to create an MD5 digest for any BAM or FASTQ files created.  ", common=true)
     public boolean CREATE_MD5_FILE = Defaults.CREATE_MD5;
 
-    @Option(shortName = StandardOptionDefinitions.REFERENCE_SHORT_NAME, doc = "Reference sequence file.", common = true, optional = true, overridable = true)
+    @ArgumentCollection
+    public ReferenceArgumentCollection referenceSequence = getReferenceArgumentCollection();
+
     public File REFERENCE_SEQUENCE = Defaults.REFERENCE_FASTA;
```

Comment:
Yes, I should have been more explicit about why I kept that. And I see your point (articulated in a comment below), about these possibly being clobbered, but CollectMultipleMetrics sets REFERENCE_SEQUENCE directly for all of the CLPs that it uses. I've reduced the visibility to protected, and added a setReferenceSequence method to SinglePassSam so that CollectMultipleMetrics can populate this, which reduces the scope of the exposure. Let me know if you have another suggestion.

On a somewhat unrelated note: in GATK4, we had previously refactored (some) of the metrics collectors that originally came from Picard in a way thats similar to what @nh13 did [here](https://github.com/broadinstitute/picard/pull/840), only more so. For each type of metric, we factored out:

1) an accumulator class for holding the results (which is also a reducer, where applicable)
2) an argument collection class for the controlling arguments
3) the collector itself, which accepts a read and populates the accumulator

In addition to codifying the ability to set the arguments for a collector independently of it's container, it entirely separates the metrics computation from the source of the reads (which in GATK's case could be a CLP, a multiple collector CLP, a Spark CLP, or a Spark RDD in a pipeline). The CLPs are just thin shells around these components. Its obsolete now since once this Barclay PR goes through we'll use the Picard version of the collectors, but I would lobby that we adopt a similar approach similar in Picard.